test/conftest.py

```diff
@@ -38,6 +33,27 @@ def client(pytestconfig):
     return SigstoreClient(entrypoint, identity_token)
 
 
+@pytest.fixture
+def make_materials_by_type():
+    def _make_materials_by_type(
+        input_name: str, cls: VerificationMaterials
+    ) -> Tuple[Path, VerificationMaterials]:
+        input_path = Path(input_name)
+        output = cls.from_input(input_path)
+
+        return (input_path, output)
+
+    return _make_materials_by_type
+
+
+@pytest.fixture(params=[BundleMaterials, SignatureCertificateMaterials])
+def make_materials(request, make_materials_by_type):
+    def _curry(input_name: str):
+        return make_materials_by_type(input_name, request.param)
+
+    return _curry
```

Comment:
```suggestion
    def _make_materials(input_name: str):
        return make_materials_by_type(input_name, request.param)

    return _make_materials
```

Let's avoid CS theory names when possible; a reader doesn't need to know what `curry` means to see that we're doing function specialization/partial evaluation 🙂